lib/OpenQA/WebAPI/Plugin/YAMLRenderer.pm

```diff
@@ -44,33 +44,33 @@ sub register {
         # validation of the schema itself which can be useful for development and testing.
         # Returns an array of errors found during validation or otherwise an empty array.
         validate_yaml => sub {
-            my ($self, $yaml, $validate_schema) = @_;
-
-            # Note: Using the schema filename; slurp'ed text isn't detected as YAML
-            my $schema_yaml = $self->app->home->child('public', 'schema', 'JobTemplate.yaml')->to_string;
-            my $validator   = JSON::Validator->new;
+            my ($self, $yaml, $schema_filename, $validate_schema) = @_;
+            my $validator = JSON::Validator->new;
             my $schema;
             my @errors;
+
             try {
+                die "No valid schema specified\n" unless ($schema_filename // '') =~ /\.yaml$/;
+                # Note: Using the schema filename; slurp'ed text isn't detected as YAML
+                my $schema_abspath = $self->app->home->child('public', 'schema', $schema_filename)->to_string;
+
                 if ($validate_schema) {
                     # Validate the schema: catches errors in type names and definitions
-                    $validator = $validator->load_and_validate_schema($schema_yaml);
+                    $validator = $validator->load_and_validate_schema($schema_abspath);
                     $schema    = $validator->schema;
                 }
                 else {
-                    $schema = $validator->schema($schema_yaml);
+                    $schema = $validator->schema($schema_abspath);
                 }
             }
             catch {
-                push @errors, $_;
+                # The last line of the backtrace gives us the error message we want
+                push @errors, (split /\n/, $_)[0];
```

Comment:
The comment says "last line", the code says first

Reply:
D'oh. The code is right in this instance.